api/src/components/crosslink/controller.ts

```diff
@@ -152,10 +157,37 @@ export const getPublicationCrosslinks = async (
             return response.json(404, { message: 'Publication not found.' });
         }
 
-        const crosslinks = await crosslinkService.getPublicationCrosslinks(
-            event.pathParameters.publicationId,
-            event.queryStringParameters.order || undefined
-        );
+        const ownLinks = event.queryStringParameters.own === 'true';
+
+        if (ownLinks && !event.user) {
```

Comment:
For these two combinations you're checking for, are they prevented in the UI? Therefore these checks are just incase someone manipulates the API request?

Reply:
For now, pretty much yes. Generally on octopus I've been putting this kind of check in at both the API & UI level. The API is starting to get a bit more standalone now so I'm mainly trying to ensure that no validation slip ups make it through. Even though something like this is unlikely to be used without the UI, I'm just following that model everywhere to be safe.